staker/block_validator.go

```diff
@@ -560,15 +561,15 @@ func (v *BlockValidator) createNextValidationEntry(ctx context.Context) (bool, e
 	}
 	var comm espressoTypes.Commitment
 	if v.config().Espresso {
-		height := endGS.HotShotHeight
-		fetchedCommitment, err := v.hotShotReader.L1HotShotCommitmentFromHeight(height)
+		_, jst, err := arbos.ParseEspressoMsg(msg.Message)
```

Comment:
Can we get the proof height by `hotshotHeight`? I don't think parsing the messages again is a good idea because here block validator is not actually validating, it is preparing the data for validating. Since we have already write the hotshot height as the part of the L2 global state, I think we can just ust it if possible

Reply:
You mean the commitment? We cant because the commitment height no longer corresponds to the current block height, but rather, a light client snapshot corresponding to some future height (hence the need to store L1ProofHeight in the justification)